### PR TITLE
Number of reports 

### PR DESCRIPTION
**Pull request by SusanaPerezJ**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java [hunk 1]
```diff
@@ -71,15 +81,15 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+          new Marker(lat, lng, crime, crimeDate, crimeTime, crimeAddress, crimeDescription, gridRow, gridCol);
       storeMarker(marker);
     } catch (Error error) {
       System.out.println(error);
     }
   }
 
   /** Fetches markers from Datastore. */
-  private List<Marker> getMarkers(HttpServletRequest request){
+  public List<Marker> getMarkers(HttpServletRequest request){
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java:92]: make this static

File: portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+public class NumberOfReportsServlet extends HttpServlet {
+  private static final Gson gson = new Gson();
+  private static final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  //List<Grid> reportsInGrids = new ArrayList<Grid>();
+  double lat;
+  double lng;
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    MarkerServlet markerServlet = new MarkerServlet();
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java:37]: don't ever create an instance of a servlet yourself. if you make the other method static, then you can just do MarkerServlet.getMarkers(request)

File: portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    List<Marker> markers = markerServlet.getMarkers(request);
+    List<Grid> reportsInGrids = new ArrayList<Grid>();
+    reportsInGrids = getNumOfReportsforAll(reportsInGrids, markers);
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java:40]: don't get it for the whole map. the caller will supply you with row and col

request.getParameter("gridrow")
request....("gridcol")

File: portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    String json = gson.toJson(reportsInGrids); //returns the number of reports for all grids
+    response.setContentType("application/json");
+    response.getWriter().println(json);    
+  }
+
+  //for each marker in the arraylist, get num of reports
+  public List<Grid> getNumOfReportsforAll(List<Grid> reportsInGrids, List<Marker> markers){
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java:47]: the List<grid> isn't necessary. just take the gridrow and gridcol that you're looking for

File: portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    for(Marker marker : markers){
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java:48]: I'm not sure what this is trying to do. it should just be 
`return markers.stream().filter(marker => marker.getGridRow() == gridrow && maker.getGridCol() == gridcol).count();`
  
you will need to delete/backfill any markers that don't have grids on them. Or you can change `getMarkers()` to not return any markers that doesn't have gridrow and gridcol

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -95,6 +164,34 @@ function createMap() {
   typeOfSearch('type-establishment', ['establishment']);
 }
 
+//prints all element on the arraylist
+function getReports(){
+  fetch('/numberOfReports')
```
Comment (wait321) [inline portfolio/src/main/webapp/script.js:169]: you don't need this

File: portfolio/src/main/webapp/script.js [hunk 7]
```diff
@@ -95,6 +164,34 @@ function createMap() {
...
+    });
+   })
+  .catch((error) => {
+    console.error(error);
+  });
+}
+
+//prints only the number of reports of a specific grid
+/*function getReportsForGrid(){
+  fetch("/numberOfReports?row=3&col=11")
```
Comment (wait321) [inline portfolio/src/main/webapp/script.js:184]: this is what we want. just make sure the servlet works with some hard coded values

File: portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java (hunk 4 above)
Comment (SusanaPerezJ) [reply portfolio/src/main/java/com/google/sps/servlets/NumberOfReportsServlet.java:47]: is the markers List unnecessary as well? or should I just delete the reportsInGrids list?

File: portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java [hunk 8]
```diff
@@ -32,11 +34,15 @@
...
   private static final String ENTITY_PROPERTY_KEY_6 = "address";
   private static final String ENTITY_PROPERTY_KEY_7 = "description";
+  private static final String ENTITY_PROPERTY_KEY_8 = "row";
+  private static final String ENTITY_PROPERTY_KEY_9 = "col";
   private static final double METERS_IN_A_MILE = 1609.34;
-  private static final Double LAT_NORTH_LIMIT = 31.747628;
-  private static final Double LAT_SOUTH_LIMIT = 31.730684;
-  private static final Double LNG_WEST_LIMIT = -106.494043;
-  private static final Double LNG_EAST_LIMIT = -106.473825;
+  private static final Double LAT_NORTH_LIMIT = 31.676131;
```
[the remark this answers is not in the document]
Comment (SusanaPerezJ) [reply portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java:40]: I made them public static final and tried to call them from the Grid class but I am getting the "cannot find symbol" error, is there a specific syntax I should use?

**Pull request merged**